src/content/reference/rules/index.md

```diff
@@ -0,0 +1,38 @@
+---
+title: Rules of React
+---
+
+<Intro>
+JavaScript rendering libraries and frameworks like React have constraints or "rules" to make the programming model cohesive and easy to reason about, while also helping you prevent bugs in your code. The rules also have the added benefit of creating a safe space for React to optimize and run your code more efficiently. This page lists all the Rules of React.
+</Intro>
+
+---
+
+These constraints are known as the **Rules of React**. They are rules – and not just guidelines – in the sense that if they are broken, your app likely has bugs. Your code also becomes unidiomatic and harder to understand and reason about.
+
+We strongly recommend using Strict Mode alongside React's ESLint plugin to help your codebase follow the Rules of React. By following the Rules of React, you'll be able to find and address these bugs, as well as prepare your codebase to work out of the box with the upcoming compiler.
+
+<DeepDive>
+#### Why are rules necessary in React? {/*why-are-rules-necessary-in-react*/}
+
+You can think of React's constraints like the grammatical rules and patterns of languages: they constrain what we can do with words, so that we can correctly and efficiently communicate our thoughts. These rules have been used in the design of all of React's features over the years. React's Strict Mode enforces several of these rules at runtime in DEV mode, and with the release of React's upcoming compiler, more rules will now be statically checked to help you find more bugs as well as allow for correct optimisation of your code.
+
+The upcoming compiler automatically makes writing simple and intuitive React code run efficiently by default. It understands JavaScript semantics and relies on the Rules of React in order to correctly and precisely optimise your codebase.
+
+However, while the compiler can detect most cases of Rules of React breakages, because of the dynamic and flexible nature of JavaScript, it is not possible to exhaustively detect all edge cases. Future iterations of the compiler will continue to improve its static detection, but it's important to understand and follow the Rules of React so that your code continues to run correctly and efficiently even if it's not possible to statically detect.
+
+If the Rules of React are broken, at best the upcoming compiler will skip optimising the components that broke the rules; and at worst, if the breakage is not statically detectable the compiled code may break in unexpected ways.
```

Comment:
I know we've iterated on this deep dive a bunch, but it doesn't really answer the question posed in the heading: why the rules are necessary. There's a strong emphasis on React Compiler here that could mislead people into thinking that the rules exist to make the compiler possible. Instead, it seems better to remove references to the compiler (or just lightly mention "the upcoming React Compiler will also depend on these rules"). And then focus more on the core issues of predictability, ability to start/stop, local reasoning, etc.

Reply:
Yeah I think you're right, it's too compiler centric right now. Let me take another pass at cleaning this up, or I might consider dropping this deep dive section altogether since it doesn't really add all that much to the page